src/state/state.F90

```diff
@@ -0,0 +1,18 @@
+! Copyright (C) 2022 National Center for Atmospheric Research
+! SPDX-License-Identifier: Apache-2.0
+!
+!> \file
+!> Aerosol model state
+
+!> The state_t type and related functions
+module aero_state
+
+  implicit none
+  private
+
+  public :: state_t
+
+  type, abstract :: state_t
```

Comment:
Do we need anything to add a finalizer here? Or if you do it on an extending class, does it Just Work(tm)?

Reply:
yeah, extending classes can add finalizers even if the base class doesn't have one (and I'm not sure if it's even possible to add a finalizer to an abstract class - it's only argument is a non-polymorphic variable)